### DISCUSSION TITLE
Extending the MsgmergeAddon

### DISCUSSION THREAD

**Discussion by lovetox** [WeblateOrg/weblate#7442] [Ideas]
Hi,

When we use Weblate our gettext flow is usually.

- Source Code gets changed
- use `xgettext`to extract new strings into a `.pot` file
- run msgmerge to update the `.po` files with the new strings

Now the last step is covered by the MsgmergeAddon

I wonder if there is interest in extending it with the xgettext step. Extending because we can’t run addons in a specific order right now with weblate (at least i saw open feature requests for that)

The proposal would be to
- add a switch that enables xgettext extraction before updating the .po files
- add a formfield where people can define paths and filenames (glob syntax) where they want to extract translations on source repo update.
- add a textfield where one can add args which are passed to xgettext

Incomplete Example:
![image](https://user-images.githubusercontent.com/8661218/160627697-a35068de-d462-49b1-afb2-9671bafe26fa.png)

Right now we have the xgettext step running in a .git hook in the weblate repo, it works but needs to be setup by a server admin for each new project.

How do other projects which use gettext for translations deal with this?

**Comment by nijel** [comment]
In many cases it's not only xgettext used for extraction, but other tools are involved as well. For example intltool or Django template extraction. That's why extraction is currently not covered...

**Reply by lovetox** [reply]
Ok i understand, so how about simply extending it with a possibility to call a script before merging translations?

Maybe leveraging code from the BaseScriptAddon

**Reply by nijel** [reply]
Running an untrusted script is generally a not good idea security wise. To do this properly, Weblate would have to integrate some kind of CI - a way to run untrusted code in containers. And I believe there are already enough CI services available...

**Reply by lovetox** [reply]
How is it untrusted? The Server Admin of the Weblate installation puts it there ..

You mention wrapping scripts with the ScriptAddon in your documentation.

**Reply by nijel** [reply]
In that case yes. I was thinking about genetic solution for this and that would involve user provided scripts
